### ISSUE TITLE
warning: loading in progress, circular require considered harmful

### ISSUE THREAD

**Issue by AlexWayfer** [getsentry/sentry-ruby#1036]
**Describe the bug**

I see warnings about circular require in `sentry-raven`.

**To Reproduce**

`require 'sentry-raven'` and enable warnings.

**Expected behavior**

 Nothing.

**Actual behavior**

```
/home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/sentry-raven-3.1.0/lib/raven/transports/http.rb:3: warning: /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/sentry-raven-3.1.0/lib/raven/transports/http.rb:3: warning: loading in progress, circular require considered harmful - /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/sentry-raven-3.1.0/lib/raven/transports.rb
	from /home/alex/.rbenv/versions/2.7.1/bin/bundle:23:in  `<main>'
	from /home/alex/.rbenv/versions/2.7.1/bin/bundle:23:in  `load'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/bundler-2.1.4/exe/bundle:34:in  `<top (required)>'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/bundler-2.1.4/lib/bundler/friendly_errors.rb:123:in  `with_friendly_errors'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/bundler-2.1.4/exe/bundle:46:in  `block in <top (required)>'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/bundler-2.1.4/lib/bundler/cli.rb:24:in  `start'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/bundler-2.1.4/lib/bundler/vendor/thor/lib/thor/base.rb:476:in  `start'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/bundler-2.1.4/lib/bundler/cli.rb:30:in  `dispatch'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/bundler-2.1.4/lib/bundler/vendor/thor/lib/thor.rb:399:in  `dispatch'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/bundler-2.1.4/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in  `invoke_command'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/bundler-2.1.4/lib/bundler/vendor/thor/lib/thor/command.rb:27:in  `run'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/bundler-2.1.4/lib/bundler/cli.rb:476:in  `exec'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/bundler-2.1.4/lib/bundler/cli/exec.rb:28:in  `run'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/bundler-2.1.4/lib/bundler/cli/exec.rb:63:in  `kernel_load'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/bundler-2.1.4/lib/bundler/cli/exec.rb:63:in  `load'
	from /home/alex/.rbenv/versions/2.7.1/bin/rspec:23:in  `<top (required)>'
	from /home/alex/.rbenv/versions/2.7.1/bin/rspec:23:in  `load'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/exe/rspec:4:in  `<top (required)>'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/runner.rb:45:in  `invoke'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/runner.rb:71:in  `run'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/runner.rb:89:in  `run'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/runner.rb:115:in  `run_specs'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/reporter.rb:74:in  `report'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/runner.rb:116:in  `block in run_specs'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/configuration.rb:2058:in  `with_suite_hooks'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/runner.rb:121:in  `block (2 levels) in run_specs'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/runner.rb:121:in  `map'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/runner.rb:121:in  `block (3 levels) in run_specs'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/example_group.rb:607:in  `run'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/example_group.rb:607:in  `map'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/example_group.rb:607:in  `block in run'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/example_group.rb:606:in  `run'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/example_group.rb:640:in  `run_examples'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/example_group.rb:640:in  `map'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/example_group.rb:644:in  `block in run_examples'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/example.rb:254:in  `run'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/example.rb:503:in  `with_around_and_singleton_context_hooks'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/example.rb:460:in  `with_around_example_hooks'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/hooks.rb:481:in  `run'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/hooks.rb:619:in  `run_around_example_hooks_for'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/hooks.rb:481:in  `block in run'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/example.rb:460:in  `block in with_around_example_hooks'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/example.rb:503:in  `block in with_around_and_singleton_context_hooks'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/example.rb:256:in  `block in run'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/example.rb:497:in  `run_before_example'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/hooks.rb:479:in  `run'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/hooks.rb:609:in  `run_example_hooks_for'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/hooks.rb:609:in  `reverse_each'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/hooks.rb:610:in  `block in run_example_hooks_for'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/hooks.rb:523:in  `run_owned_hooks_for'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/hooks.rb:523:in  `each'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/hooks.rb:524:in  `block in run_owned_hooks_for'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/hooks.rb:360:in  `run'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/example.rb:450:in  `instance_exec'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/example.rb:450:in  `instance_exec'
	from /home/alex/Projects/stream_timer/spec/web/index_spec.rb:7:in  `block (2 levels) in <top (required)>'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/memoized_helpers.rb:317:in  `block in let'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/memoized_helpers.rb:155:in  `fetch_or_store'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/memoized_helpers.rb:155:in  `fetch'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/memoized_helpers.rb:156:in  `block in fetch_or_store'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-support-3.9.3/lib/rspec/support/reentrant_mutex.rb:23:in  `synchronize'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/memoized_helpers.rb:157:in  `block (2 levels) in fetch_or_store'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/memoized_helpers.rb:157:in  `fetch'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/memoized_helpers.rb:157:in  `block (3 levels) in fetch_or_store'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/memoized_helpers.rb:317:in  `block (2 levels) in let'
	from /home/alex/Projects/stream_timer/spec/web/spec_helper.rb:14:in  `block (2 levels) in <top (required)>'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/memoized_helpers.rb:317:in  `block in let'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/memoized_helpers.rb:155:in  `fetch_or_store'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/memoized_helpers.rb:155:in  `fetch'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/memoized_helpers.rb:156:in  `block in fetch_or_store'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-support-3.9.3/lib/rspec/support/reentrant_mutex.rb:23:in  `synchronize'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/memoized_helpers.rb:157:in  `block (2 levels) in fetch_or_store'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/memoized_helpers.rb:157:in  `fetch'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/memoized_helpers.rb:157:in  `block (3 levels) in fetch_or_store'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/rspec-core-3.9.2/lib/rspec/core/memoized_helpers.rb:317:in  `block (2 levels) in let'
	from /home/alex/Projects/stream_timer/spec/web/spec_helper.rb:9:in  `block (2 levels) in <top (required)>'
	from /home/alex/Projects/stream_timer/spec/web/spec_helper.rb:9:in  `require_relative'
	from /home/alex/Projects/stream_timer/config/main.rb:3:in  `<top (required)>'
	from /home/alex/Projects/stream_timer/config/main.rb:3:in  `require_relative'
	from /home/alex/Projects/stream_timer/application.rb:5:in  `<top (required)>'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/bundler-2.1.4/lib/bundler.rb:174:in  `require'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/bundler-2.1.4/lib/bundler/runtime.rb:58:in  `require'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/bundler-2.1.4/lib/bundler/runtime.rb:58:in  `each'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/bundler-2.1.4/lib/bundler/runtime.rb:65:in  `block in require'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/bundler-2.1.4/lib/bundler/runtime.rb:88:in  `rescue in block in require'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/bundler-2.1.4/lib/bundler/runtime.rb:88:in  `require'
	from /home/alex/Projects/stream_timer/lib/flame/raven_context.rb:5:in  `<top (required)>'
	from /home/alex/Projects/stream_timer/lib/flame/raven_context.rb:5:in  `require'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/sentry-raven-3.1.0/lib/sentry-raven.rb:1:in  `<top (required)>'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/sentry-raven-3.1.0/lib/sentry-raven.rb:1:in  `require'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/sentry-raven-3.1.0/lib/raven.rb:1:in  `<top (required)>'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/sentry-raven-3.1.0/lib/raven.rb:1:in  `require'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/sentry-raven-3.1.0/lib/raven/base.rb:15:in  `<top (required)>'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/sentry-raven-3.1.0/lib/raven/base.rb:15:in  `require'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/sentry-raven-3.1.0/lib/raven/client.rb:7:in  `<top (required)>'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/sentry-raven-3.1.0/lib/raven/client.rb:7:in  `require'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/sentry-raven-3.1.0/lib/raven/transports.rb:18:in  `<top (required)>'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/sentry-raven-3.1.0/lib/raven/transports.rb:18:in  `require'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/sentry-raven-3.1.0/lib/raven/transports/http.rb:3:in  `<top (required)>'
	from /home/alex/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/sentry-raven-3.1.0/lib/raven/transports/http.rb:3:in  `require'
```

**Environment**
 - Ruby Version: 2.7.1
 - SDK Version: 3.1.0

**Comment by st0012** [comment]
this has been fixed in https://github.com/getsentry/sentry-ruby/pull/1035 and will be released in version `3.1.1` this week

**Comment by AlexWayfer** [comment]
> this has been fixed in #1035 and will be released in version `3.1.1` this week

Oh, OK, thanks. I looked for issues by warning text.

**Comment by st0012** [comment]
np, we always welcome such reports 😉
